### ISSUE TITLE
Error 'vips__write: write failed' for some character images

### ISSUE THREAD

**Issue by jarmoza** [printprobability/printprob-db#82]
Certain characters in the book `redo/anon_R7329_columbia_4_miraclesnoviolationsREDO1683` are not appearing on the [character grouping page](https://printprobdb.psc.edu/group_characters?book=0378fecd-40d4-48fb-88e4-eb48597c9691&order=character_class&character_run=f83bfb1c-2ea3-43af-b926-3d3ad473b1ec&character_class=B_uc&show_damaged_characters=false) and produce the following error when accessed directly via [API url](https://printprobdb.psc.edu/iiif/books/redo/anon_R7329_columbia_4_miraclesnoviolationsREDO1683/pages_color/anon_R7329_columbia_4_miraclesnoviolationsREDO1683-0001.tif/950,4000,62,152/full/0/default.jpg).

```
vips__write: write failed
unix error: No space left on device
```

This appears to have to do with the large memory sizes of the images of this book (130-140MB), and may be related to this memory buffer issue as noted in [this issue](https://github.com/lovell/sharp/issues/1661) of the `sharp` NodeJS image processing library.

The dev's [suggested fix](https://github.com/lovell/sharp/issues/1661#issuecomment-484696125) is to up the value assigned to environment variable `VIPS_DISC_THRESHOLD` which defaults to 100MB.

**Comment by jarmoza** [comment]
Update on this issue.

The thought from the issue above and this issue https://github.com/microsoft/hi-ml/issues/628 is that setting one of the two environmental variables `DATASET_MOUNT_CACHE_SIZE` or `VIPS_DISC_THRESHOLD`. So far this has still resulted in the same error.

These are passed to the singularity instance via the `--env` argument to the command in `startup.sh`

They are evidently being set though it's not clear if the disc threshold changes anything relevant to the error in question. Altering the mount cache size variable to 300GB as suggested resulted in character images not being output correctly on the workbench.

I have set up virtual box/vagrant on my laptop in order to edit the singularity instance's runscript to check the values of the environment variables in the IIIF singularity instance. Apparently, they are not set at all nor are any IIIF-related environment variables.

It's still likely that the mount cache size variable adjustment may help the error in question. Another possibility is adjust the memory scratch space variable for the go-iiif-vips library.

It does not appear memory values are adjustable in the iiif config.json we have on our server (this is documented in the go-iiif repo)

**Comment by jarmoza** [comment]
While looking into possibilities of why the `iiif-server` program would be producing this `vips` error, I noticed the config settings in this commit in our `shared/iiif-server/config/config.json` is actually missing.

https://github.com/printprobability/printprob-db/commit/46f080354110c583535c15059eed60f046cab3fe

Matt had labelled this commit "aggressive caching". As seen here in the `go-iiif` [documentation](https://github.com/go-iiif/go-iiif/blob/main/README.md#config-files):

```
ttl is the maximum number of seconds an image should live in cache.
limit the maximum number of megabytes the cache should hold at any one time.
```

Re-establishing Matt's 8000 seconds with 500mb cache doesn't seem to have fixed the error though.

**Comment by jarmoza** [comment]
The next step will be to alter my local copy of the `go-iiif-vips` singularity instance with the runscript altered to have large VIPS_DISC_THRESHOLD environment variable set. This will be temp swapped in for our regular `go-iiif-vips` singularity instance.

**Comment by jarmoza** [comment]
This issue is now fixed. Chalk it up a ChatGPT hallucination, the original idea to up `VIPS_DISC_THRESHOLD` above 100mb was correct, but environment variables are inserted via the `singularity run` command and not via the `singularity start` command.

I am unsure as to why building a new container based off of the old one but with an environment variable set in its runscript caused the IIIF server not to deliver images – and should we need to alter the runscript in the future this remains a concern.

The `VIPS_DISC_THRESHOLD` value of `250m` will be added to the startup instructions so it runs as such:

`singularity run --env VIPS_DISC_THRESHOLD=250m instance://iiif1 > iiif.log &`

This README will be updated in the repo and as noted [above](https://github.com/printprobability/printprob-db/issues/82#issuecomment-2072728662), the current config file will be committed to the repo as well.

**Comment by jarmoza** [comment]
Closed by 235b7d0448086c6d93337739461c1cb81a39af93 and 3529402701e625b54a2786e40a8397c2b9e950ae.
